File: swing-live/swinglive.apk_Decompiler.com/sources/com/swing/live/StreamForegroundService.java
package com.swing.live;

import J3.i;
import android.app.Notification;
import android.app.NotificationChannel;
import android.app.NotificationManager;
import android.app.PendingIntent;
import android.app.Service;
import android.content.Intent;
import android.os.Build;
import android.os.IBinder;
import k.AbstractC0501s;
import q.m;

/* JADX INFO: loaded from: classes.dex */
public final class StreamForegroundService extends Service {

    /* JADX INFO: renamed from: a, reason: collision with root package name */
    public static final /* synthetic */ int f3872a = 0;

    @Override // android.app.Service
    public final IBinder onBind(Intent intent) {
        return null;
    }

    @Override // android.app.Service
    public final void onCreate() {
        super.onCreate();
        if (Build.VERSION.SDK_INT >= 26) {
            AbstractC0501s.m();
            NotificationChannel notificationChannelB = AbstractC0501s.b();
            notificationChannelB.setDescription("Notification for active live streaming sessions");
            ((NotificationManager) getSystemService(NotificationManager.class)).createNotificationChannel(notificationChannelB);
        }
    }

    @Override // android.app.Service
    public final int onStartCommand(Intent intent, int i4, int i5) {
        PendingIntent activity = PendingIntent.getActivity(this, 0, new Intent(this, (Class<?>) MainActivity.class), 201326592);
        m mVar = new m(this, "swing_live_streaming");
        mVar.e = m.b("SwingLive: Streaming active");
        mVar.f6227f = m.b("Broadcasting match to YouTube...");
        Notification notification = mVar.f6237p;
        notification.icon = android.R.drawable.ic_menu_camera;
        mVar.f6229h = 1;
        mVar.f6233l = "service";
        mVar.f6228g = activity;
        notification.flags |= 2;
        Notification notificationA = mVar.a();
        i.d(notificationA, "build(...)");
        startForeground(888, notificationA);
        return 1;
    }
}

File: swing-live/android/app/src/main/kotlin/com/dhandha/swing/swing_live/StreamForegroundService.kt
package com.dhandha.swing.swing_live

import android.app.Notification
import android.app.NotificationChannel
import android.app.NotificationManager
import android.app.PendingIntent
import android.app.Service
import android.content.Context
import android.content.Intent
import android.content.pm.ServiceInfo
import android.os.Build
import android.os.IBinder
import androidx.core.app.NotificationCompat
import androidx.core.app.ServiceCompat

class StreamForegroundService : Service() {
    private val CHANNEL_ID = "swing_live_streaming"
    private val NOTIFICATION_ID = 888

    override fun onBind(intent: Intent?): IBinder? = null

    override fun onCreate() {
        super.onCreate()
        createNotificationChannel()
    }

    override fun onStartCommand(intent: Intent?, flags: Int, startId: Int): Int {
        val notificationIntent = Intent(this, MainActivity::class.java)
        val pendingIntent = PendingIntent.getActivity(
            this, 0, notificationIntent,
            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) PendingIntent.FLAG_IMMUTABLE else 0
        )

        val notification: Notification = NotificationCompat.Builder(this, CHANNEL_ID)
            .setContentTitle("SwingLive: Streaming active")
            .setContentText("Broadcasting match to YouTube...")
            .setSmallIcon(android.R.drawable.ic_menu_camera)
            .setContentIntent(pendingIntent)
            .setOngoing(true)
            .setPriority(NotificationCompat.PRIORITY_HIGH)
            .build()

        // Android 14+ requires the FGS type to be passed explicitly to
        // startForeground; otherwise the service crashes with
        // MissingForegroundServiceTypeException the moment we call
        // startService while the app is anywhere near a background
        // transition. ServiceCompat takes care of the version split.
        val type = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.UPSIDE_DOWN_CAKE) {
            ServiceInfo.FOREGROUND_SERVICE_TYPE_CAMERA or
                ServiceInfo.FOREGROUND_SERVICE_TYPE_MICROPHONE
        } else {
            0
        }
        ServiceCompat.startForeground(this, NOTIFICATION_ID, notification, type)
        return START_STICKY
    }

    private fun createNotificationChannel() {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
            val serviceChannel = NotificationChannel(
                CHANNEL_ID,
                "SwingLive Streaming Channel",
                NotificationManager.IMPORTANCE_DEFAULT
            ).apply {
                description = "Notification for active live streaming sessions"
            }
            val manager = getSystemService(NotificationManager::class.java)
            manager.createNotificationChannel(serviceChannel)
        }
    }
}
